src/components/Frontend.tsx

```diff
@@ -1,82 +1,65 @@
 "use client";
 
-import { useEffect, useCallback, useState } from "react";
-import sdk, {
-  FrameNotificationDetails,
-  type FrameContext,
-} from "@farcaster/frame-sdk";
+import { useEffect, useCallback, useState, useRef } from "react";
+import sdk, { type FrameContext } from "@farcaster/frame-sdk";
 import { useAccount, useDisconnect, useConnect, useChainId } from "wagmi";
 
 import { config } from "~/components/providers/WagmiProvider";
 import { Button } from "~/components/ui/Button";
-import { useRequestAttestation } from "~/hooks/useRequestAttestation";
+import { useReportSybil } from "~/hooks/useReportSybil";
+import { useUserData } from "~/hooks/useUserData";
+
+import ReCAPTCHA from "react-google-recaptcha";
 
 export default function Frontend(
   { title }: { title?: string } = { title: "Check Sybil" }
 ) {
   const [isSDKLoaded, setIsSDKLoaded] = useState(false);
   const [context, setContext] = useState<FrameContext>();
-
   const [added, setAdded] = useState(false);
-  const [notificationDetails, setNotificationDetails] =
-    useState<FrameNotificationDetails | null>(null);
-
-  const [lastEvent, setLastEvent] = useState("");
-
   const [addFrameResult, setAddFrameResult] = useState("");
-
-  useEffect(() => {
-    setNotificationDetails(context?.client.notificationDetails ?? null);
-  }, [context]);
+  const [searchIdentifier, setSearchIdentifier] = useState<string>("");
+  const [currentIdentifier, setCurrentIdentifier] = useState<
+    number | string | null
+  >(null);
```

Comment:
IIUC this identifier can be the username or the fid, right? I think that it is ok to only accept the username since it is more user-friendly IMO. I am not sure if it is possible to exist numeric usernames that would be confused with the FID 🤔